pkg/apis/crd/v1alpha1/types.go

```diff
@@ -591,3 +596,25 @@ type TierList struct {
 
 	Items []Tier `json:"items"`
 }
+
+// PeerServiceAccounts can use NamespacedName, ObjectSelector or
+// NamespacedObjectSelector to select ServiceAccounts.
+type PeerServiceAccounts struct {
+	NamespacedName           *NamespacedName           `json:"namespacedName,omitempty"`
+	ObjectSelector           *metav1.LabelSelector     `json:"objectSelector,omitempty"`
+	NamespacedObjectSelector *NamespacedObjectSelector `json:"namespacedObjectSelector,omitempty"`
```

Comment:
I'm not sure we need both of these. Do you think just `NamespacedObjectSelector` could be sufficient (with a `nil` `NamespaceSelector` being equivalent to just the `ObjectSelector`, i.e. all Namespaces)?

Reply:
Yes, just `NamespacedObjectSelector` is sufficient. I was thinking if this design is more in line with the verbose API idea. I'm ok with both. Let's see what's @Dyanngg 's thoughts.

Reply:
Changed.